### PR TITLE
OvmfPkg: Rerun dispatcher after initializing virtio-rng

### PR DESCRIPTION
**Pull request by osteffenrh**
# Description
Since the pixiefail CVE fix the network stack requires a hardware random number generator. This can currently be a modern CPU supporting the RDRAND instruction or a virtio-rng device.
The latter is initialized during the BDS phase.
To ensure all depending (network) modules are also started, we need to run the dispatcher once more.
Without this, network boot is not available under certain hardware configurations.

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Run Ovmf in Qemu with a core2duo vCPU, a virtio-rng-pci device and a varying number of isa serial ports. Observe if the network stack is loaded (presence of HTTP/PXE boot entries, for example). With 2 serial ports the netboot is not available.

## Integration Instructions
N/A

### CODE REVIEW

**Comment by osteffenrh** [comment]
^ @kraxel @stefano-garzarella

**Comment by osteffenrh** [comment]
mailing list thread: https://edk2.groups.io/g/devel/topic/109008158#msg120623

File: OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c [hunk 1]
```diff
@@ -539,6 +539,8 @@ PlatformBootManagerBeforeConsole (
     ConnectVirtioPciRng,
     NULL
     );
+
+  gDS->Dispatch ();
```
Comment (stefano-garzarella) [inline OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c:543]: Should we include this fix also in `ConnectVirtioPciRng()` in `OvmfPkg/Library/PlatformBootManagerLibBhyve/BdsPlatform.c` ?

Comment (osteffenrh) [reply OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c:543]: Probably a good idea. I'll open a separate PR for that later.

**Comment by stefano-garzarella** [comment]
What about adding `Fixes: 4c4ceb2ceb ("NetworkPkg: SECURITY PATCH CVE-2023-45237")` in the commit description?
Not sure if it's usual in edk2, but it looks like there is some use of it. I think it will help backports.

**Comment by kraxel** [comment]
@mdkinney can this be included in edk2-stable202411 please?

**Comment by kraxel** [comment]
> What about adding `Fixes: 4c4ceb2ceb ("NetworkPkg: SECURITY PATCH CVE-2023-45237")` in the commit description? Not sure if it's usual in edk2, but it looks like there is some use of it. I think it will help backports.

Some people do it (mostly the ones which are used to doing this from other projects I think), but it is not universally used practice in edk2.  So nice to have for sure, but not a hard requirement.

**Comment by osteffenrh** [comment]
I am adding it now

> > What about adding `Fixes: 4c4ceb2ceb ("NetworkPkg: SECURITY PATCH CVE-2023-45237")` in the commit description? Not sure if it's usual in edk2, but it looks like there is some use of it. I think it will help backports.
> 
> Some people do it (mostly the ones which are used to doing this from other projects I think), but it is not universally used practice in edk2. So nice to have for sure, but not a hard requirement.

I am adding it now.

**Review by ardbiesheuvel — APPROVED** [review]

**Pull request merged**